src/controllers/OrganisationsController.js

```diff
@@ -22,6 +23,413 @@ function validateAndRender (res, name, params) {
   return render(res, name, schema, params)
 }
 
+/**
+ * Middleware. Attempts to fetch data and short-circuits with 404 when
+ * data for given query does not exist. Meant to be used to fetch singular records.
+ *
+ * `this` needs `{ query({ req, params }) => any, result: string }`
+ *
+ * where the `result` is the key under which result of the query will be stored in `req`
+ *
+ * @param {*} req
+ * @param {*} res
+ * @param {*} next
+ */
+async function fetchOne (req, res, next) {
+  logger.debug({ type: types.App, message: 'fetchOne', resultKey: this.result })
+  const query = this.query({ req, params: req.params })
+  const result = await datasette.runQuery(query)
+  if (result.formattedData.length === 0) {
+    // we can make the 404 more informative by informing the use what exactly was "not found"
+    res.status(404).render('errorPages/404', { })
```

Comment:
completely agree with this, might be worth raising a ticket and putting it in the chat. tag Alex and charlie to see what they think

Reply:
(even if its only in non production environments)